test-cypress/integration/home/language.change.spec.js

```diff
@@ -0,0 +1,44 @@
+import HomeSteps from '../../steps/home-steps';
+import SparqlSteps from '../../steps/sparql-steps';
+
+describe('Home screen language validation', () => {
+    let repositoryId;
+
+    beforeEach(() => {
+        cy.viewport(1280, 1000);
+        HomeSteps.visitAndWaitLoader();
+        repositoryId = '23repo' + Date.now();
+    });
+
+    afterEach(() => {
+        cy.deleteRepository(repositoryId);
+        // Change the language back to English
+        SparqlSteps.changeLanguage('Anglais');
+    });
+
+    context('Language change label checks', () => {
+        it('Default language should be active and language change should affect labels', () => {
+            cy.createRepository({id: repositoryId});
+            cy.initializeRepository(repositoryId);
+            cy.enableAutocomplete(repositoryId);
+            cy.presetRepository(repositoryId);
+
+            // When I visit home page with selected repository
+            HomeSteps.visitAndWaitLoader();
+            // Check some labels are in default language
```

Comment:
You need to select a concrete element and check for its content. The reason is that `cy.contains('some text')` actually finds the first element in the DOM which contains the text and it might not be the one you expected. So basically you shuld have something like
```
HomeSteps.getViewResourceAsTableButton().should('contain', 'Table');
```
And in the HomeSteps:
```
static getViewResourceAsTableButton() {
  return cy.get('.display-type-table-btn');
}
```

Same applies for the rest of the labels you need to verify. If concrete elements in the application don't have proper css classes or ids, the you can add some in order to be able to select them in the tests.

Reply:
I understand. I'll see what I can do here. I think I'll need to add some ids.